### PR TITLE
Updated the dependencies.

### PR DESCRIPTION
**Pull request by bvitaliyg**
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/build.gradle [hunk 1]
```diff
@@ -35,6 +35,10 @@ android {
     }
   }
   defaultPublishConfig "ossRelease"
+  compileOptions {
```
Comment (bvitaliyg) [inline Backpack/build.gradle:38]: This is needed for better binary compatibility with the dependencies.

File: Backpack/build.gradle [hunk 2]
```diff
@@ -44,13 +48,14 @@ dependencies {
...
   implementation "androidx.cardview:cardview:$rootProject.ext.cardView"
+  implementation "androidx.swiperefreshlayout:swiperefreshlayout:$rootProject.ext.swiperefreshLayout"
   implementation "com.jakewharton.threetenabp:threetenabp:$rootProject.ext.threeTenAbp"
-  androidTestImplementation "androidx.test.ext:junit:1.1.1"
-  androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
-  androidTestImplementation 'androidx.test:rules:1.2.0'
-  androidTestImplementation 'com.nhaarman.mockitokotlin2:mockito-kotlin:2.1.0'
-  androidTestImplementation 'org.mockito:mockito-android:2.28.2'
-  testImplementation 'junit:junit:4.13.1'
+  androidTestImplementation "androidx.test.ext:junit:$rootProject.ext.junitAndroid"
```
Comment (bvitaliyg) [inline Backpack/build.gradle:53]: I have extracted the version numbers to constants.

File: Backpack/build.gradle [hunk 3]
```diff
@@ -44,13 +48,14 @@ dependencies {
   implementation "com.google.android.material:material:$rootProject.ext.androidMaterial"
   implementation "androidx.constraintlayout:constraintlayout:$rootProject.ext.constraintLayout"
   implementation "androidx.cardview:cardview:$rootProject.ext.cardView"
+  implementation "androidx.swiperefreshlayout:swiperefreshlayout:$rootProject.ext.swiperefreshLayout"
```
Comment (bvitaliyg) [inline Backpack/build.gradle:51]: SwipeToRefresh (we use modified progress indicator from there) was extracted to separate library.

File: app/src/androidTest/java/net/skyscanner/backpack/toast/BpkToastTests.kt [hunk 4]
```diff
@@ -37,13 +38,13 @@ class BpkToastTests : BpkSnapshotTest() {
   fun screenshotTestToast_Default() {
     val toast = BpkToast.makeText(testContext, "Test", BpkToast.LENGTH_SHORT)
     toast.show()
-    snap(toast.view)
+    snap(toast.view!!)
```
Comment (bvitaliyg) [inline app/src/androidTest/java/net/skyscanner/backpack/toast/BpkToastTests.kt:41]: With compileSdk=30 Toast.view was deprecated. The docs states it was done to stop the developers to put custom views in toasts.

However, we don't use it in this manner. We only need it to take a photo of Toast for tests ,ad it's working fine.

File: app/src/main/java/net/skyscanner/backpack/demo/ComponentDetailActivity.kt [hunk 5]
```diff
@@ -52,6 +52,7 @@ class ComponentDetailActivity : BpkBaseActivity() {
       // Create the detail fragment and add it to the activity
       // using a fragment transaction.
       val itemId = intent.getStringExtra(ComponentDetailFragment.ARG_ITEM_ID)
+        ?: error("ComponentDetailActivity intent must have ${ComponentDetailFragment.ARG_ITEM_ID} set")
```
Comment (bvitaliyg) [inline app/src/main/java/net/skyscanner/backpack/demo/ComponentDetailActivity.kt:55]: Nullability information was added to Intent.getStringExtra() in combileSdk=30. Therefore we need to do a nullability check here.

File: app/src/main/java/net/skyscanner/backpack/demo/ThemeApplier.kt [hunk 6]
```diff
@@ -25,14 +25,14 @@ import net.skyscanner.backpack.demo.data.SharedPreferences
...
-  override fun onActivityPaused(activity: Activity?) {}
-  override fun onActivityResumed(activity: Activity?) {}
-  override fun onActivityStarted(activity: Activity?) {}
-  override fun onActivityDestroyed(activity: Activity?) {}
-  override fun onActivitySaveInstanceState(activity: Activity?, outState: Bundle?) {}
-  override fun onActivityStopped(activity: Activity?) {}
-  override fun onActivityCreated(activity: Activity?, savedInstanceState: Bundle?) {
-    activity?.theme?.applyStyle(SharedPreferences.getTheme(activity), true)
-    activity?.let { BpkTheme.applyDefaultsToContext(it) }
+  override fun onActivityPaused(activity: Activity) {}
```
Comment (bvitaliyg) [inline app/src/main/java/net/skyscanner/backpack/demo/ThemeApplier.kt:28]: Nullability info was added to ActivityLifecycleCallbacks. It states Activity is never null in these methods.

File: app/src/main/res/layout/component_settings_theme_option.xml [hunk 7]
```diff
@@ -20,6 +21,6 @@
     android:layout_height="wrap_content"
     android:src="@drawable/bpk_tick"
     android:layout_weight=".2"
-    android:tint="?attr/colorPrimary"
+    app:tint="?attr/colorPrimary"
```
Comment (bvitaliyg) [inline app/src/main/res/layout/component_settings_theme_option.xml:24]: Lint complains about this. New approach is the official way it needs to be written. It's also more compatible.

**Comment by bvitaliyg** [comment]
The reason why the screenshots were updated is because in new versions there's a change in RTL behaviour of NavBar:
image
(green version is new).

I've checked our code – it seems we always use equals distances for LTR and RTL NavBars. They have probably fixed some bug in the library related to incorrect RTL spacings which caused the diff.

**Review by georgegillams — APPROVED** [review]
LGTM 👍

**Pull request merged**